h2o-core/src/main/java/water/api/RequestServer.java

```diff
@@ -314,12 +313,18 @@ public void doGeneric(String method, HttpServletRequest request, HttpServletResp
 
       resp.writeTo(response.getOutputStream());
 
+    } catch (IllegalArgumentException e) {  // This error type should be passed to clients.
```

Comment:
Just tried running it in Chrome:

```
This site can’t be reached

The webpage at http://localhost:54321/3/Models.java/GLRM_model_python_1502900833427_1 might be temporarily down or it may have moved permanently to a new web address.
ERR_INVALID_RESPONSE
```

You are on the right track, I am just not sure if singling out IAE here is the right way of doing it.